### PR TITLE
Add C++20 three-way comparison operator

### PR DESCRIPTION
**Pull request by Farigh**
The new operator, also known as the [spaceship operator](https://en.cppreference.com/w/cpp/language/punctuators#.3C.3D.3E_.28since_C.2B.2B20.29), was introduced in the C++20 standard.

The NSIQ parser now handles it properly.

---

I fixed some tests as well in this PR and added a GitHub Action to ensure this is performed for each PR from now on.
I greatly encourage you to enforce Checks on PR merge, this can be found in the branch protection settings : 
image

### CODE REVIEW

**Comment by Farigh** [comment]
Here is what the Test workflow looks like : 
image

**Comment by Farigh** [comment]
@kunaltyagi do you plan on merging this at some point by any chance?

**Comment by mosherubin** [comment]
@Farigh I'm a contributor, couldn't find your email, drop me a line at moshe.rubin@gmail.com

File: nsiqcppstyle_checker.py [hunk 1]
```diff
@@ -1229,13 +1230,13 @@ def RunRules(ruleManager, lexer):
...
-                if t.type == 'TYPE':
+                if t.type == "TYPE":
                     ruleManager.RunTypeNameRule(lexer, t.value.upper(), t.fullName,
                                                 t.decl, t.contextStack, t.context)
-                elif t.type == 'FUNCTION':
+                elif t.type == "FUNCTION":
                     ruleManager.RunFunctionNameRule(lexer, t.fullName, t.decl,
                                                     t.contextStack, t.context)
-                elif ((t.type == 'COMMENT') or (t.type == 'CPPCOMMENT')):
+                elif t.type in ["COMMENT", "CPPCOMMENT"]:
```
Comment (kunaltyagi) [inline nsiqcppstyle_checker.py:1239]:
```suggestion
                elif t.type in ("COMMENT", "CPPCOMMENT"):
```

**Review by kunaltyagi — APPROVED** [review]
Thanks for the neat update to tests @Farigh

**Pull request merged**